src/ui/map.js

```diff
@@ -1032,9 +1032,14 @@ class Map extends Camera {
     }
 
     /**
-     * Updates the map's Mapbox style object with a new value. If a style already is set and options.diff is true,
-     * this compares the style against the map's current state and performs only the changes necessary to make
-     * the map style match the desired state.
+     * Updates the map's Mapbox style object with a new value.
+     *
+     * If a style is already set when this is used and options.diff is set to true, the map renderer will attempt to compare the given style
```

Comment:
It would be useful to add why it is valuable to enable to diff. If a style is not changing drastically, it enables a cross-faded change from the current style to the next and it is more performant - by reusing existing source and layer definitions and transitioning them to a future state - it saves on CPU cycles, and possibly network bandwidth.